### PR TITLE
Checking role of the user to authorize requests 

### PR DESCRIPTION
**Pull request by o-bm**
Hi all,
couple of things to note 

- I changed checkrole so that it is only required on admin routes. normal routes that both admin and user can access are not affected. default value for a new user role is still user. 
- checkrole now fetches the user from the utorid in the request and checks the user's role. 
- To have a test admin user just pass in admin as utorid and admin@example.com as email

I also tested it out and it works 
Here is a request to a protected route 
image
Here is an admin making a request to that same route 
image

### CODE REVIEW

**Comment by o-bm** [comment]
close #49

File: server/routes/CommentRoutes.js [hunk 1]
```diff
@@ -1,19 +1,13 @@
...
-const { AsyncWrapController } = require("../routes/ErrorHandlingMiddleware");
+const { AsyncWrapController } = require("../middleware/ErrorHandlingMiddleware");
+const checkRole = require("../middleware/CheckRoleMiddleware");
 
 AsyncWrapController(Comment);
 
-// Get a comment for a solution.
 router.get("/:solutionId", Comment.get);
-
-// Create a new comment for a solution in the database.
```
Comment (Vladmidir) [inline server/routes/CommentRoutes.js:10]: why did u delete comments?

File: server/routes/UserRoutes.js [hunk 2]
```diff
@@ -1,19 +1,12 @@
...
-
-// Create a new user in the database.
-router.post("/:id", User.create);
-
-// Update a user in the database.
-router.put("/:id", User.update);
-
-// Delete a User from the database.
-router.delete("/:id", User.delete);
+router.post("/:id", checkRole(['admin']), User.create);
```
Comment (Vladmidir) [inline server/routes/UserRoutes.js:8]: pls keep the spacing 🙏

File: server/routes/ErrorHandlingMiddleware.js
Comment (Vladmidir) [inline server/routes/ErrorHandlingMiddleware.js:1]: why delete?

File: server/routes/LoggingMiddleware.js
Comment (Vladmidir) [inline server/routes/LoggingMiddleware.js:1]: why delete this one too?

File: server/routes/CommentRoutes.js (hunk 1 above)
Comment (o-bm) [reply server/routes/CommentRoutes.js:10]: the route is well defined. I don't see the point of comments there

File: server/routes/CommentRoutes.js [hunk 3]
```diff
@@ -1,19 +1,13 @@
...
 router.post("/:id", Comment.create);
-
-// Edit a comment for a solution in the database.
 router.put("/:id", Comment.edit);
-
-// Delete a comment for a solution from the database.
-router.delete("/:id", Comment.delete);
+router.delete("/:id", checkRole(['admin']), Comment.delete);
```
Comment (Vladmidir) [inline server/routes/CommentRoutes.js:11]: double and single quotes in one line

**Review by Vladmidir — COMMENTED** [review]
Added some comments you should address.

I also suggest you rebase off main. I have cleaned up the backend so it should be a lot easier to work with. Remember to `force:true` the db.

File: server/routes/CommentRoutes.js (hunk 1 above)
Comment (Vladmidir) [reply server/routes/CommentRoutes.js:10]: pls keep it

File: server/routes/ErrorHandlingMiddleware.js
Comment (o-bm) [reply server/routes/ErrorHandlingMiddleware.js:1]: must have been by accident

File: server/routes/LoggingMiddleware.js
Comment (o-bm) [reply server/routes/LoggingMiddleware.js:1]: must have been by accident

File: server/routes/CommentRoutes.js (hunk 3 above)
Comment (o-bm) [reply server/routes/CommentRoutes.js:11]: wil update

File: server/routes/ErrorHandlingMiddleware.js
Comment (o-bm) [reply server/routes/ErrorHandlingMiddleware.js:1]: they are moved to middleware directory in previous PR

File: server/index.js [hunk 4]
```diff
@@ -1,17 +1,50 @@
+const session = require('express-session');
 const express = require("express");
 const path = require("node:path");
 const db = require("./models");
 const importChallenges = require("./scripts/importChallenges");
 const { ErrorHandler } = require("./middleware/ErrorHandlingMiddleware");
 const loggingMiddleware = require("./middleware/LoggingMiddleware");
+<<<<<<< HEAD
```
Comment (arthurhohinng) [inline server/index.js:8]: Did not merge this correctly as it is showing the conflicts here

File: server/index.js [hunk 5]
```diff
@@ -1,17 +1,50 @@
...
+const checkRole = require("./middleware/CheckRoleMiddleware");
+>>>>>>> 8775bfc ( added testing for check role)
 
 const app = express();
 
 app.use(express.json());
 
 const PORT = process.env.PORT || 8080;
 
+<<<<<<< HEAD
```
Comment (arthurhohinng) [inline server/index.js:21]: Same here

**Review by arthurhohinng — CHANGES_REQUESTED** [review]
Please rebase correctly first as the commit is showing the conflicts that aren't resolved

**Comment by o-bm** [comment]
done

**Pull request merged**